src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -59,6 +60,22 @@ public int getRange() {
         return RANGE;
     }
 
+    @Nonnull
+    public Map<Location, EnergyNetProvider> getGenerators() {
+        return Collections.unmodifiableMap(generators);
+    }
+    
+    @Nonnull
+    public Map<Location, EnergyNetComponent> getCapacitors() {
+        return Collections.unmodifiableMap(capacitors);
+    }
+    
+    @Nonnull
+    public Map<Location, EnergyNetComponent> getConsumers() {
+        return Collections.unmodifiableMap(consumers);
+    }
+
+    @Nonnull
     @Override
     public String getId() {
```

Comment:
Formatting. Also API methods should always have javadocs.
```suggestion
    public @Nonnull Map<Location, EnergyNetProvider> getGenerators() {
        return Collections.unmodifiableMap(generators);
    }
    
    public @Nonnull Map<Location, EnergyNetComponent> getCapacitors() {
        return Collections.unmodifiableMap(capacitors);
    }
    
    public @Nonnull Map<Location, EnergyNetComponent> getConsumers() {
        return Collections.unmodifiableMap(consumers);
    }

    @Override
    public @Nonnull String getId() {
```